### ISSUE TITLE
val_accuracy and val_loss not visible during model training even after adding validation data

### ISSUE THREAD

**Issue by Raghav2305** [zekaouinoureddine/COVID-19-DETECTION-FROM-X-RAY#3]
I am training a ResNet 50 model to classify images as Covid infected or normal. While training, the only metrics that are shown are training accuracy and loss. The val_accuracy and val_loss are not visible in any epochs. I have passed the validation data but it's still not showing up.

```
X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=0.2)
print('The train Data Shape ', X_train.shape[0])
 
X_test, X_valid, y_test, y_valid = train_test_split(X_test,y_test, test_size = 0.5)
print('The validation Data Shape ', X_valid.shape[0])
print('The test Data Shape ', X_test.shape[0])

```

```
lr = 1e-3
epochs = 50
bs = 8
optimizer = Adam(learning_rate = lr, decay= lr/epochs)
model.compile(optimizer, loss='binary_crossentropy', metrics=['accuracy'])

epochs = 100

#initialize the training data augmentation object
train_datagen = ImageDataGenerator(
        rotation_range=15,
        fill_mode ="nearest")
metric = "accuracy"
 
checkpointer = ModelCheckpoint(filepath = "./Model/CDX_Best_RestNet50.h5", save_best_only = True, monitor = metric , verbose=1)
start = time.time()
 
# let's get started !
 
history=model.fit(train_datagen.flow(X_train, y_train, batch_size = bs),
                            steps_per_epoch = len(X_train)//bs,
                            validation_data = (X_valid, y_valid),
                            validation_steps = len(X_valid)//bs,
                            epochs =epochs,
                            callbacks= [checkpointer])
 
end = time.time()
duration = end - start
print ('\n This Model took %0.2f seconds (%0.1f minutes) to train for %d epochs'%(duration, duration/60, epochs) )

```

**Also, this is the shape of X_train: The train Data Shape (224, 224, 3)**
**The validation metrics are visible only for the first epoch**:

![image](https://user-images.githubusercontent.com/82498222/232193347-b63e13e2-ca07-439e-89c3-c7635d38b6fa.jpg)

**Comment by zekaouinoureddine** [comment]
If you implement the fixes for issue #2 in your code, you should be able to see the `val_accuracy` and `val_loss` metrics for all epochs. Note, I used only 4 epochs with a small number of training and validation examples to reproduce the issue. As shown in the image below, the suggested changes in #2 successfully resolved this issue as well.

---
![image](https://user-images.githubusercontent.com/61702091/232266367-b3fb948b-54ad-4d84-8f59-e4bfda5d5c14.png)
